cpp/tests/utilities/cudf_test_utils.cuh

```diff
@@ -365,24 +365,38 @@ bool gdf_equal_columns(gdf_column* left, gdf_column* right)
   if (left->null_count != right->null_count) return false;
   if (left->dtype_info.time_unit != right->dtype_info.time_unit) return false;
 
-  if (!(left->data && right->data))
+  if ((left->data == nullptr) != (right->data == nullptr))
     return false;  // if one is null but not both
 
-  if (!thrust::equal(thrust::cuda::par, reinterpret_cast<T*>(left->data),
+  if ((left->data != nullptr) && 
+      !thrust::equal(thrust::cuda::par, reinterpret_cast<T*>(left->data),
                      reinterpret_cast<T*>(left->data) + left->size,
                      reinterpret_cast<T*>(right->data)))
     return false;
 
-  if (!(left->valid && right->valid))
+  if ((left->valid == nullptr) != (right->valid == nullptr))
     return false;  // if one is null but not both
 
-  if (!thrust::equal(thrust::cuda::par, left->valid,
+  if ((left->valid != nullptr) && 
+      !thrust::equal(thrust::cuda::par, left->valid,
```

Comment:
This `thrust::equal` is incorrect. It will erroneously compare bits in the bitmask that do not refer to actual elements if `col->size % sizeof(bit_mask_t) != 0`.

Reply:
You can see how I handle this correctly in `column_wrapper::operator==`: https://github.com/rapidsai/cudf/blob/2f7c0d2464cac5c9acab2dd683dda69f3b049ffb/cpp/tests/utilities/column_wrapper.cuh#L382

I think the implementation in `column_wrapper::operator==` should be refactored into this function, and then `column_wrapper::operator==` can just call this function.

Reply:
Done.